pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,182 @@
+    description: Date and time when the message was sent.
+ ```
+Here the expected payload `lightMeasured` is an integer. A validation error will be included if the producer sends a message with the type boolean.
+ ```
+ "_asyncapi_eg_validation_error": "{\"ts\":\"2021-12-20T11:33:26.583143572Z\",\"errors\":[\"lumens: Invalid type. Expected: integer, given: boolean\"]}",
+ ```
+## Validation handled by the broker
+As producers and consumers do not communicate with each other directly, but rather information transfer happens via Kafka topic. At the same time, the consumer still needs to know the type of data the producer is sending. Imagine if the producer starts sending bad data to Kafka or if the data type of your data gets changed. We need a way to have a common data type that must be agreed upon.
+
+That’s where Schema Registry comes into the picture. It is an application that resides outside of your Kafka cluster and handles the distribution of schemas to the producer and consumer by storing a copy of schema in its local cache.
+
+```mermaid
+graph LR
+    A([Producer]) --->|send schemaID| B(Schema Registry) & C{Kafka Cluster}--->|recieve schemaID| D([Consumer])
+```
+With the schema registry in place, the producer first talks to the schema registry and checks if the schema is available before sending the data to Kafka. If it cannot locate the schema, it registers and caches it in the schema registry. When the producer receives the schema,to sends it to Kafka prefixed with a unique schema ID. When the consumer processes this message, it will communicate with the schema registry using the schema ID obtained from the producer. If there is a schema mismatch, the schema registry will throw an error, informing the producer that it is violating the schema agreement.
+
+```mermaid
+stateDiagram-v2
+    Producer --> Broker : messages
+    Producer --> Fail : error
+    Broker --> Schema_registry : validation request
+    Schema_registry --> consumer : valid message
+    Schema_registry --> Fail : invalid message
+```
+
+Broker validation can also be implemented using other methods. AsyncAPI is not involved in validation here but the schemas that are stored in Schema registry can be referenced from AsyncAPI documents.
+Here's an example of AsyncAPI document where you can see `schemaFormat` and `payload` 
+```
+asyncapi: 2.0.0
+info:
+  title: Example with Avro
+  version: 0.1.0
+channels:
+  example:
+    publish:
+      message:
+        schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
+        payload:
+          $ref: 'https://example.europe-west3.gcp.confluent.cloud/subjects/test/versions/1/schema'
+```
+
+# Additional Resources
+- Check out an [AsyncAPI file demo with Studio.](https://studio.asyncapi.com/?url=https://raw.githubusercontent.com/asyncapi/event-gateway/master/deployments/k8s/event-gateway-demo/event-gateway-demo.asyncapi.yaml)
+
+- Check out other [AsyncAPI validators](https://www.asyncapi.com/docs/tools#validators)
```

Comment:
not good resource as this is a mix of message/AsyncAPI document and other validators

Reply:
added asyncapi demo in gateway section and removed the last one. Any suggestions for resources? or just remove the section

Reply:
I don't think there are any